src/runtime/lib/telemetry/index.ts

```diff
@@ -85,62 +103,112 @@ export class TelemetryClient {
      */
-    constructor(options: TelemetryClientOptions) {
-        const { } = options;
+    private constructor(options?: TelemetryClientOptions) {
+        if (options) {
+            const { } = options;
+        }
         this.httpClient = axios.create({ timeout: 3000, headers: { 'Content-Type': 'text/plain' } });
         this.serverUrl = 'https://client-telemetry.amazonalexa.com';
         this.postRetries = 3;
         this.enabled = this.isEnabled();
+        const osType = os.type();
+        const osArch = os.arch();
+        const osRelease = os.release();
+        const gitVersionUint8Array: Uint8Array = child_process.execSync(
+            'git --version');
+        const gitVersion = new TextDecoder(DEFAULT_ENCODING).decode(gitVersionUint8Array);
         this.data = {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
             version: extensions.getExtension(EXTENSION_ID)?.packageJSON.version,
             machineId: env.machineId,
             timeStarted: new Date(),
             newUser: false,
             timeUploaded: null,
             clientId: EXTENSION_FULL_NAME,
+            operatingSystem: `${osType} ${osArch} ${osRelease}`,
+            dependencyVersion: version,
+            gitVersion: gitVersion,
             actions: []
         };
     }
 
+    /**
+     * The static method that controls the access to the TelemetryClient instance
+     * @param options 
+     * @returns 
+     */
+    public static getInstance(options?: TelemetryClientOptions) {
+        if (TelemetryClient.instance == undefined) {
+            TelemetryClient.instance = new TelemetryClient(options);
+        }
+        return TelemetryClient.instance;
+    }
+
     /**
      * Starts action
      * @param {string} name - The action name
      * @param {string} type - The action type
      * @return {MetricAction}
      */
-    public startAction(name: string, type: string): MetricAction{
-        const action = new MetricAction(name, type);
-        this.data.actions.push(action);
-        return action;
+    public startAction(name: string, type: ActionType): MetricAction {
+        return new MetricAction(name, type);
     }
 
     /**
-     * Sends data to the metric server
-     * @param {Error} [error] - Error object indicating an error.
+     * Store an action and send the data
+     * @param action - The MetricAction item
+     * @param error - The error object
+     */
+    public async store(action: MetricAction , error?:Error): Promise<void> {
+        action.end(error);
+        if (ext.context.globalState.get(TELEMETRY_UPDATE_TIME) == undefined) {
+            const now = new Date().getTime();
+            ext.context.globalState.update(TELEMETRY_UPDATE_TIME, now);
+        }
+        if (ext.context.globalState.get(TELEMETRY_DATA) == undefined) {
+            ext.context.globalState.update(TELEMETRY_DATA, [action]);
+            return;
```

Comment:
Why are we returning back the process to the caller here, instead of calling `sendData` ?

Reply:
Because the new `action` has been added in the the globalState, and the globalState data just initialize, so it will not be sent out (less than 15 mins).
 
To avoid confusion, removed `action` in line 184 and removed `return` in line 185